kie-api/src/main/java/org/kie/api/io/ResourceType.java

```diff
@@ -60,12 +64,25 @@ public static ResourceType addResourceTypeToRegistry(final String resourceType,
                                                          final String defaultExtension,
                                                          final String... otherExtensions) {
 
+        return addResourceTypeToRegistry(resourceType, true, description, defaultPath, defaultExtension, otherExtensions);
+    }
+
+    public static ResourceType addResourceTypeToRegistry(final String resourceType,
+                                                         boolean isNative,
+                                                         final String description,
+                                                         final String defaultPath,
+                                                         final String defaultExtension,
+                                                         final String... otherExtensions) {
+
         ResourceType resource = new ResourceType( resourceType,
+                                                  isNative,
                                                   description,
                                                   defaultPath,
                                                   defaultExtension,
                                                   otherExtensions );
+
         CACHE.put( resourceType, resource );
+        resource.getAllExtensions().forEach( ext -> CACHE.put( "." + ext, resource ) );
```

Comment:
Is this change related with the PR or it is just an unrelated fix?

Reply:
Not related and not a fix, just a minor performance improvement to avoid that it loops all the available resource types when trying to determine the type of a file.

Reply:
Ok thanks for the clarification :+1: